qt/aqt/browser/browser.py

```diff
@@ -560,7 +560,7 @@ def setupSidebar(self) -> None:
 
         # schedule sidebar to refresh after browser window has loaded, so the
         # UI is more responsive
-        self.mw.progress.timer(10, self.sidebar.refresh, False, parent=self.sidebar)
+        self.mw.progress.single_shot(10, self.sidebar.refresh)
```

Comment:
This (and other similar cases below) will now throw an error if the widget is closed before the timer has a chance to fire, so we're introducing a bit of a regression here. Perhaps we should be reserving singe_shot() for cases where no appropriate parent exists? Any code that uses them should probably be defensively written, eg testing sip.is_deleted() before doing what it wants to do, and we could mention that in the docstring.

Reply:
That's what I meant with problem 1. Before #1672, all timers were parented to mw, so also weren't dropped if the widget got destroyed in the meantime.